libimage/filters.go

```diff
@@ -51,13 +58,25 @@ func (i *Image) applyFilters(ctx context.Context, filters compiledFilters, tree
 // filterImages returns a slice of images which are passing all specified
 // filters.
 // tree must be provided if compileImageFilters indicated it is necessary.
-func (r *Runtime) filterImages(ctx context.Context, images []*Image, filters compiledFilters, tree *layerTree) ([]*Image, error) {
+func (r *Runtime) filterImages(ctx context.Context, images []*Image, filters compiledFilters, referenceFilter referenceFilterFunc, tree *layerTree) ([]*Image, error) {
```

Comment:
This works, but every user needs to manage `referenceFilter` separately.

Consider turning `compiledFilters` into a `struct` that carries everything.